components/formats-bsd/src/loci/formats/out/TiffWriter.java

```diff
@@ -545,7 +552,17 @@ else if (retrieve.getPixelsBinDataCount(series) == 0) {
   @Override
   public int getTileSizeX() throws FormatException {
     if (tileSizeX == 0) {
-      tileSizeX = super.getTileSizeX();
+      MetadataOptions options = getMetadataOptions();
+      boolean defaultTileWriting = TILE_WRITING_DEFAULT;
+      if (options instanceof FormatOptions) {
+        defaultTileWriting = ((FormatOptions) options).getBoolean(DEFAULT_TILE_WRITING_KEY, TILE_WRITING_DEFAULT); 
```

Comment:
Why don't we get a `TILE_SIZE_X` property here, and use that if non-null?  The "defaultTileWriting" logic and key doesn't seem to be conveying the needed information (the user-specified tile size).  The same applies to the Y tile size.